test/unit/api_spec.js

```diff
@@ -2260,132 +2073,110 @@ describe("api", function () {
+      const loadingTask = getDocument(transport);
+      const pdfDocument = await loadingTask.promise;
+      expect(pdfDocument.numPages).toEqual(14);
 
-          return pdfDocument.getPage(10);
-        })
-        .then(function (pdfPage) {
-          expect(pdfPage.rotate).toEqual(0);
-          expect(fetches).toBeGreaterThan(2);
+      const pdfPage = await pdfDocument.getPage(10);
+      expect(pdfPage.rotate).toEqual(0);
+      expect(fetches).toBeGreaterThan(2);
 
-          loadingTask.destroy().then(done);
-        })
-        .catch(done.fail);
+      await loadingTask.destroy();
     });
 
-    it("should fetch document info and page using range and streaming", function (done) {
+    it("should fetch document info and page using range and streaming", async function () {
       const initialDataLength = 4000;
-      let fetches = 0,
-        loadingTask;
+      let fetches = 0;
+
+      const data = await dataPromise;
+      const initialData = data.subarray(0, initialDataLength);
+      const transport = new PDFDataRangeTransport(data.length, initialData);
+      transport.requestDataRange = function (begin, end) {
+        fetches++;
+        if (fetches === 1) {
+          // Send rest of the data on first range request.
+          transport.onDataProgressiveRead(data.subarray(initialDataLength));
+        }
+        waitSome(function () {
+          transport.onDataRange(begin, data.subarray(begin, end));
+        });
+      };
 
-      dataPromise
-        .then(function (data) {
-          const initialData = data.subarray(0, initialDataLength);
-          const transport = new PDFDataRangeTransport(data.length, initialData);
-          transport.requestDataRange = function (begin, end) {
-            fetches++;
-            if (fetches === 1) {
-              // Send rest of the data on first range request.
-              transport.onDataProgressiveRead(data.subarray(initialDataLength));
-            }
-            waitSome(function () {
-              transport.onDataRange(begin, data.subarray(begin, end));
-            });
-          };
-          loadingTask = getDocument(transport);
-          return loadingTask.promise;
-        })
-        .then(function (pdfDocument) {
-          expect(pdfDocument.numPages).toEqual(14);
+      const loadingTask = getDocument(transport);
+      const pdfDocument = await loadingTask.promise;
+      expect(pdfDocument.numPages).toEqual(14);
 
-          return pdfDocument.getPage(10);
-        })
-        .then(function (pdfPage) {
-          expect(pdfPage.rotate).toEqual(0);
-          expect(fetches).toEqual(1);
+      const pdfPage = await pdfDocument.getPage(10);
+      expect(pdfPage.rotate).toEqual(0);
+      expect(fetches).toEqual(1);
 
-          waitSome(function () {
-            loadingTask.destroy().then(done);
-          });
-        })
-        .catch(done.fail);
+      waitSome(async function () {
+        await loadingTask.destroy();
```

Comment:
Are you *absolutely sure* that this works as intended, since it looks like the test will now finish *before* destruction is complete. I'd expect that what you actually want is something like:
```javascript
await new Promise(resolve => {
  waitSome(resolve);
});
await loadingTask.destroy();
```

Reply:
Good point; fixed now.